test/Test_Synthesizer.ml

```diff
@@ -39,8 +39,74 @@ let y_MINUS_x_MINUS_z_LE_x () =
     constants = []
   } in Alcotest.(check string) "identical" "(not (or (= w x) (= y z)))" result.string
 
+let select () =
+  let open Synthesizer in
+  let result = solve ~config:{ Config.default with logic = Logic.of_string "ALIA" } {
+    arg_names = [ "a" ; "k" ];
+    inputs = [ (Array.map ~f:(fun (a,b,c,d) -> Value.Array (a,b,c,d))
+                         [| (Type.INT, Type.INT, [ (Value.Int 3, Value.Int 30)
+                                                 ; (Value.Int 2, Value.Int 20)
+                                                 ; (Value.Int 1, Value.Int 10) ],
+                                                 Value.Int 1)
+                          ; (Type.INT, Type.INT, [ (Value.Int 2, Value.Int 20)
+                                                 ; (Value.Int 1, Value.Int 1024) ],
+                                                 Value.Int 0)
+                          ; (Type.INT, Type.INT, [(Value.Int 0 , Value.Int 0)], Value.Int 30) |])
+             ; [| Value.Int 1 ; Value.Int 2 ; Value.Int 3 |] ];
+    outputs = [| Value.Int 10 ; Value.Int 20 ; Value.Int 30 |];
+    constants = []
+  } in Alcotest.(check string) "identical" "(select a k)" result.string
+
+let store () =
+let open Synthesizer in
+let result = solve ~config:{ Config.default with logic = Logic.of_string "ALIA" } {
+  arg_names = [ "a" ; "k" ; "v"];
+  inputs = [ (Array.map ~f:(fun (a,b,c,d) -> Value.Array (a,b,c,d))
+                        [| (Type.INT, Type.INT, [ ],
+                                                Value.Int 1)
+                        ; (Type.INT, Type.INT, [ ],
+                                                Value.Int 0)
+                        ; (Type.INT, Type.INT, [], Value.Int 30) |])
+            ; [| Value.Int 1 ; Value.Int 2 ; Value.Int 3 |]
+            ; [| Value.Int 20 ; Value.Int 40 ; Value.Int 6 |]
+            ];
+  outputs = [| Value.Array((Type.INT, Type.INT, [ (Value.Int 1, Value.Int 20) ],
+                                                  Value.Int 1))
+              ; Value.Array((Type.INT, Type.INT, [ (Value.Int 2, Value.Int 40)],
+                                                   Value.Int 0))
+              ; Value.Array((Type.INT, Type.INT, [ (Value.Int 3 , Value.Int 6)], 
+                                                    Value.Int 30))|];
+  constants = []
+} in Alcotest.(check string) "identical" "(store a k v)" result.string
```

Comment:
The semicolons `;` don't seem to be properly aligned

Reply:
Check and let me know if this needs to be changed